pkg/controller/mysqlbackup/internal/syncer/job.go

```diff
@@ -91,6 +91,10 @@ func (s *jobSyncer) SyncFn() error {
 // getBackupCandidate returns the hostname of the first not-lagged and
 // replicating slave node, else returns the master node.
 func (s *jobSyncer) getBackupCandidate() string {
+	if s.backup.Spec.CandidateNode != "" {
```

Comment:
I tend to do the validate here, and if the checks fail, we can return the original result.

Reply:
It will be better indeed.